### ISSUE TITLE
Support for featured snippets

### ISSUE THREAD

**Issue by VijayKumarHackr** [jarun/googler#139]
Display featured snippets from google search to make things convenient. Since most of the times, the best answer is in the featured snippet.
By featured snippet, I mean this: https://support.google.com/webmasters/answer/6229325

**Comment by jarun** [comment]
Hi,

Currently I don't have any plans to add new features to `googler` other than maintenance and bug fixes. If you really want to see it in `googler`, please feel free to raise a PR.

**Comment by zmwangx** [comment]
googler is designed to work with link results with a title and an abstract. I was going to explain the problems with "featured snippets" in detail, and typed up a lot of things, but then I realized I didn't know the boundaries between these "featured snippets" and other kind of ["smart cards"](https://github.com/jarun/googler/blob/8b9d205/googler#L1086-L1087) well, and didn't want to state facts wrong. The short version is (we believe) we can't support them well. If you believe it's possible with minor changes to parser logic, please open a PR with links to example pages for testing, thanks.

**Comment by VijayKumarHackr** [comment]
I'm new to open source community. I'm still learning my ways to do things with Git and GitHub. I'd love to raise a PR but it'll take some time for me to tweak and understand your code before I start contributing to it. Thank you.

**Comment by jarun** [comment]
No problem!

**Comment by zmwangx** [comment]
I want to add one practical example of why we can't support featured snippets well:

![screen shot 2016-11-06 at 11 03 34 am](https://cloud.githubusercontent.com/assets/4149852/20039379/c748669c-a410-11e6-90f0-f6efa2949b29.png)

The query here is "quark masses", and the snippet is an HTML table, which is fairly common in featured snippets. Apparently we can't display an HTML table inline without a dedicated parser (easy) and renderer (hard, and even if we have one it will probably be of dubious quality in many cases).

Also, contrary to my memory, https://github.com/jarun/googler/blob/8b9d205/googler#L1088 does not filter out the featured snippet (since there's a nested `div.g` inside the `div.g.mnr-c.g-blk`), resulting in an unsatisfactory duplicate:

```
$ googler quark masses

1 Quarks - HyperPhysics
http://hyperphysics.phy-astr.gsu.edu/hbase/particles/quark.html

2 Quark - Wikipedia
https://en.wikipedia.org/wiki/Quark
Jump to Mass - A quark (/ˈkwɔːrk/ or /ˈkwɑːrk/) is an elementary particle and a fundamental constituent of matter.
Quarks combine to form composite particles called hadrons, the most stable of which are protons and neutrons, the
components of atomic nuclei.

3 Quarks - HyperPhysics
http://hyperphysics.phy-astr.gsu.edu/hbase/particles/quark.html
With a mass of 3100 MeV, over three times that of the proton, this particle was the first example of another quark,
called the charm quark. The J/Psi is made up of a charm-anticharm quark pair. The lightest meson which contains a
charm quark is the D meson.

...
```

But at least you're not missing out on any result.

Basically, there's a limit to the expressiveness of a text-based program. Trying to flatten an HTML data source doesn't always work. Think of text-based browsers.

**Comment by zmwangx** [comment]
@jarun Based on the analysis above, I think this feature request could be safely closed. Of course, a working PR could still be considered (if it at least handles tables in a satisfactory way).

**Comment by jarun** [comment]
I agree the ROI is not at all impressive for this feature. Closing accordingly. Thanks @zmwangx and @VijayKumarHackr.

**Comment by VijayKumarHackr** [comment]
Thanks @jarun for consideration! And @zmwangx, that was very neat
explaination...

On Sun, Nov 6, 2016 at 10:02 PM, Arun Prakash Jana <notifications@github.com
